app/src/main/java/com/github/devcordde/pluginjamsystem/services/TeamService.java

```diff
@@ -1,19 +1,192 @@
+                        .build(guildId, teamId, userId)
+                )
+                .exchangeToMono(response -> {
+                    if (response.statusCode() == HttpStatus.OK) {
+                        return response.bodyToMono(LeaderToken.class);
+                    }
+                    // TODO: error handling
+                    return Mono.empty();
+                })
+                .block();
+    }
+
+    @Nullable
+    public LeaderToken isLeaderAuthValid(long guildId, int teamId, String token) {
+        return client.get().uri(uriBuilder -> uriBuilder
+                        .path("{guildId}/{teamId}/leader/auth/check")
+                        .build(guildId, teamId)
+                )
+                .header("leader-authorization", token)
+                .exchangeToMono(response -> {
+                    if (response.statusCode() == HttpStatus.OK) {
+                        return response.bodyToMono(LeaderToken.class);
+                    }
+                    // TODO: error handling
+                    return Mono.empty();
+                })
+                .block();
+    }
+
+    @Nullable
+    public TeamProfile setTeamName(long guildId, int teamId, String token, String name) {
+        var teamProfile = new TeamProfile(name, 0L, "", "");
+        return client.put().uri(uriBuilder -> uriBuilder
+                        .path("{guildId}/{teamId}/leader/name")
+                        .build(guildId, teamId)
+                )
+                .header("leader-authorization", token)
+                .bodyValue(teamProfile)
+                .exchangeToMono(response -> {
+                    if (response.statusCode() == HttpStatus.OK) {
+                        return response.bodyToMono(TeamProfile.class);
+                    }
+                    // TODO: error handling
+                    return Mono.empty();
+                })
+                .block();
+    }
+
+    @Nullable
+    public TeamProfile setTeamProjectUrl(long guildId, int teamId, String token, String projectUrl) {
+        var teamProfile = new TeamProfile(null, 0L, projectUrl, null);
+        return client.put().uri(uriBuilder -> uriBuilder
+                        .path("{guildId}/{teamId}/leader/projecturl")
+                        .build(guildId, teamId)
+                )
+                .header("leader-authorization", token)
+                .bodyValue(teamProfile)
+                .exchangeToMono(response -> {
+                    if (response.statusCode() == HttpStatus.OK) {
+                        return response.bodyToMono(TeamProfile.class);
+                    }
+                    // TODO: error handling
+                    return Mono.empty();
+                })
+                .block();
+    }
+
+    @Nullable
+    public TeamProfile setTeamLeader(long guildId, int teamId, String token, long leader) {
+        var teamProfile = new TeamProfile(null, leader, null, null);
```

Comment:
The question here is, why did you decide to pass a TeamProfile if you only allow to change the leader?

Question is targeting all occurrences.

Reply:
Thats more a decision of consistency. I want all endpoints to use json in general. I wanted to avoid using plain text. Instead of creating an object with just an value, I thought it might be enough to just send the object with null values and the value to change. This allows easy read and write on both sides. The data send for the empty keys is negligible imo